ts/src/main/scala/ch/epfl/ts/traders/BacktestTrader.scala

```diff
@@ -0,0 +1,105 @@
+package ch.epfl.ts.traders
+
+import scala.language.postfixOps
+import scala.collection.mutable.{Map => MMap, MutableList => MList}
+import scala.concurrent.duration.{DurationInt, DurationLong}
+import akka.actor.Cancellable
+import ch.epfl.ts.component.{Component, ComponentRef}
+import ch.epfl.ts.data._
+import ch.epfl.ts.data.Currency._
+
+/**
+  * Implements back testing for traders.
+  *
+  * To use this class, redirect all previous connections into trader to instances
+  * of this class. Connections out of the original trader should remain unchanged.
+  *
+  * @param trader the reference to the real trader component
+  * @param traderId the id of the trader
+  * @param initial the initial seed money
+  * @param currency currency of the intial seed money
+  * @param period the period in milliseconds to compute performance
+  */
+class BacktestTrader(trader: ComponentRef, traderId: Long, initial: Double, currency: Currency, period: Long) extends Component {
+  // for usage of scheduler
+  import context._
+
+  private var schedule: Cancellable = null
+  private val wallet = MMap[Currency, Double](currency -> initial)
```

Comment:
`Wallet` is a component in itself and should be instantiated along with the traders (we have only 1 example doing it correctly in the codebase now I think). It can then be queried to give, take money and ask how much there is.
@sygi can give details on the design decisions.